manifests/plugin/container_gateway.pp

```diff
@@ -24,11 +36,30 @@
   Stdlib::HTTPUrl $pulp_endpoint = "https://${facts['networking']['fqdn']}",
   Stdlib::Absolutepath $sqlite_db_path = '/var/lib/foreman-proxy/smart_proxy_container_gateway.db',
   Optional[Integer] $sqlite_timeout = undef,
+  Boolean $manage_postgresql = true,
+  Optional[Stdlib::Host] $postgresql_host = undef,
+  Optional[Stdlib::Port] $postgresql_port = undef,
+  String $postgresql_database = 'container_gateway',
+  String $postgresql_user = 'foreman-proxy',
+  String $postgresql_password = extlib::cache_data('container_gateway_cache_data', 'db_password', extlib::random_password(32))
 ) {
   foreman_proxy::plugin::module { 'container_gateway':
     version   => $version,
     enabled   => $enabled,
     feature   => 'Container_Gateway',
     listen_on => $listen_on,
   }
+
+  if $foreman_proxy::plugin::container_gateway::manage_postgresql {
+    include postgresql::server
+    postgresql::server::db { $foreman_proxy::plugin::container_gateway::postgresql_database:
+      user     => $foreman_proxy::plugin::container_gateway::postgresql_user,
+      password => postgresql::postgresql_password(
+        $foreman_proxy::plugin::container_gateway::postgresql_user,
+        $foreman_proxy::plugin::container_gateway::postgresql_password
```

Comment:
If we make the default to use unix socket auth, what should happen with the password? Would the postgres DB not need a password in that case? If that's the answer,  then I suppose I could conditionally create the postgresql db definition here with or without a password.

The user will need to be set I'm assuming so the database is created in relation to `foreman-proxy` by default. I think it'd be safe to include the username without the password in the connection string. Otherwise we'll need some conditional for determining if the username needs to be filled into the DB connection string.